userspace/libsinsp/sinsp.cpp

```diff
@@ -1426,25 +1423,20 @@ int32_t sinsp::next(OUT sinsp_evt **puevt)
 	// things like exit() or close() can be parsed.
 	//
 	uint32_t nfdr = (uint32_t)m_fds_to_remove->size();
-
 	if(nfdr != 0)
 	{
 		/* This is a removal logic we shouldn't scan /proc. If we don't have the thread 
 		 * to remove we are fine.
 		 */
 		sinsp_threadinfo* ptinfo = get_thread_ref(m_tid_of_fd_to_remove, false).get();
-		if(!ptinfo)
-		{
-			ASSERT(false);
-			return res;
-		}
-
-		for(uint32_t j = 0; j < nfdr; j++)
+		if(ptinfo)
```

Comment:
Here `return res` would return `SCAP_SUCCESS` but `*puevt` was not set. Avoid dropping an event altogether, since here it was not even parsed/processed.

Reply:
nice catch! :)